interp/numTok.mli

```diff
@@ -1,18 +1,88 @@
-type t = {
-  int : string;  (** \[0-9\]\[0-9_\]* *)
-  frac : string;  (** empty or \[0-9_\]+ *)
-  exp : string  (** empty or \[eE\]\[+-\]?\[0-9\]\[0-9_\]* *)
-}
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
 
-val equal : t -> t -> bool
+(** Numerals in different forms: signed or unsigned, possibly with
+    fractional part and exponent; we preserve the original user
+    representation for better error reporting *)
```

Comment:
If I remember well, it is also needed for user defined notations that would contain numerals (c.f. comment below)

https://github.com/coq/coq/blob/18aa9ca60ec9b3d1712276ec0c615dfe54c1a251/interp/constrexpr.ml#L64

Reply:
I changed the message. Tell me if you consider that it fixes the previous inaccuracy.

Reply:
I would rather move the entire comment from constrexpr.ml (which seems now a bit out of place / inaccurate) to numTok.mli.

Reply:
BTW, shouldn't we take the opportunity to add a test for this, something along the following lines
```Coq
Notation "0" := Prop.
Notation "+0" := O.
Notation "-0" := true.
Notation "00" := (nil, nil).                                                                                                                          
Check 0 : Prop.
Check +0 : nat.
Check -0 : bool.
Check 00 : prod _ _.

Notation "1000" := False.
Notation "1_000" := O.
Check 1000 : Prop.
Check 1_000 : nat.
```